python/activator/middleware_interface.py

```diff
@@ -282,21 +322,13 @@ def run_pipeline(self, visit: Visit, snaps: set) -> None:
             in the `visit` object, but we'll have to test how that works once
             we implemented this with actual data.
         """
-        pipeline = PIPELINE_MAP[visit.kind]
-        _log.info(f"Running pipeline {pipeline} on visit '{visit}', snaps {snaps}")
-        cmd = [
-            "echo",
-            "pipetask",
-            "run",
-            "-b",
-            self.butler,
-            "-p",
-            pipeline,
-            "-i",
-            f"{self.instrument}/raw/all",
-        ]
-        _log.debug("pipetask command line: %s", cmd)
-        # subprocess.run(cmd, check=True)
+        where = f"detector={visit.detector} and exposure in ({','.join(str(x) for x in snaps)})"
+        pipeline = SimplePipelineExecutor.from_pipeline(self.pipeline, where=where, butler=self.butler)
```

Comment:
`pipeline` is actually a `SimplePipelineExecutor` object... maybe call it `executor`?